miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/ads/AdMobDisplayer.kt

```diff
@@ -64,4 +82,76 @@ class AdMobDisplayer(private val context: Context) : MiniAppAdDisplayer, Corouti
                 onFailed.invoke("Ad is not loaded yet")
         }
     }
+
+    /** Load the rewarded ad when it is ready. **/
+    override fun loadRewarded(adUnitId: String, onLoaded: () -> Unit, onFailed: (String) -> Unit) {
```

Comment:
We don't have to do this now (or maybe at all), but I was thinking about something: Maybe we could use a callback with a [Kotlin Result](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/) so that we don't need two callbacks:

`override fun loadRewarded(adUnitId: String, callback: (result: Result<String?>) -> Unit)`

This more closely matches how the iOS side has done callbacks. But the problem is that it doesn't match what we've done with the other callbacks on Android, so I'm not sure if it's a good idea to have two different types of callbacks in the public interfaces.

What do you think?

Reply:
In AdMob case, I think the only callback `callback: (result: Result<String?>) -> Unit` will limit the extension capability because there are many [ad events](https://developers.google.com/admob/android/interstitial#ad_events) and we may want to implement it next time.

For the other functions just have two results: success and failure like `shareInfo`, it's a good practice.